src/puppetlabs/jdbc_util/core.clj

```diff
@@ -159,3 +160,17 @@
                        [sql 0]
                        seq-params-w-indices)]
     (vec (conj (flatten parameters) sql'))))
+
+(defn wrap-with-db-permission-error-handling
+  "A ring middleware to catch and report postgresql permission-denied errors."
+  [handler]
+  (fn [request]
+    (try
+      (handler request)
+      (catch PSQLException e
+        (if (= "42501" (.getSQLState e))
+          {:status 403
+           :headers {"Content-Type" "application/json"}
+           :body (json/generate-string {:kind "db-permission-error"
+                                        :msg (tru "Insufficient database permissions")})}
```

Comment:
We should have docs review this error string but I'm not sure whose purview this falls under, and I'm not finding Melissa or Mindy's github names via autocomplete.

But generally we try to have error messages be full sentences.